src/raw.rs

```diff
@@ -27,415 +313,572 @@ where
     }
 }
 
-pub struct Get<'a> {
-    client: &'a Client,
-    key: Key,
+impl ColumnFamily {
+    pub fn into_inner(self) -> String {
+        self.0
+    }
+}
+
+impl Deref for ColumnFamily {
+    type Target = String;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+pub trait RequestInner: Sized {
+    type Resp;
+
+    fn execute(self, client: Arc<RpcClient>, cf: Option<ColumnFamily>) -> KvFuture<Self::Resp>;
+}
+
+pub struct Request<Inner>
```

Comment:
Maybe this would be better as an enum?

```rust
pub enum Request<Inner> {
    Uninitiated { client: Arc<RpcClient>, inner: Inner, cf: Option<ColumnFamily> },
    Initiated { future: KvFuture<Inner::Resp>> },
}
```

Reply:
I have tried multiple ways to make it work. But unfortunately due to the reason that enum requires moving the old value and conflicts with `Future::poll` which takes a `&mut` reference. Give me some more time, let's see if I can figure out other way of doing it.

Reply:
I have pushed a new commit with Request changed to enum, PTAL @Hoverbear